pkg/irsa/capa/capa.go

```diff
@@ -388,40 +360,20 @@ func (s *Service) Delete(ctx context.Context) error {
 		}
 
 		data := cfConfig.Data
-		cfDistributionId = string(data["distributionId"])
-		cfOriginAccessIdentityId = string(data["originAccessIdentityId"])
-	}
-
-	err = s.IAM.DeleteOIDCProviders()
-	if err != nil {
-		ctrlmetrics.Errors.WithLabelValues(s.Scope.Installation(), s.Scope.AccountID(), s.Scope.ClusterName(), s.Scope.ClusterNamespace()).Inc()
-		s.Scope.Logger().Error(err, "failed to delete OIDC provider")
-		return err
-	}
+		cfDistributionId := string(data["distributionId"])
+		cfOriginAccessIdentityId := string(data["originAccessIdentityId"])
 
-	clusterValuesConfigMap, err := s.getClusterValuesConfigMap(ctx)
-	if err != nil {
-		return err
-	}
-
-	if !key.IsChina(s.Scope.Region()) {
 		err = s.Cloudfront.DisableDistribution(cfDistributionId)
 		if err != nil {
 			s.Scope.Logger().Error(err, "failed to disable cloudfront distribution for cluster")
 			return err
 		}
 
-		deleteDistribution := func() error {
-			err = s.Cloudfront.DeleteDistribution(cfDistributionId)
-			if err != nil {
-				return err
-			}
-			return nil
+		err = s.Cloudfront.DeleteDistribution(cfDistributionId)
```

Comment:
Instead of blocking the execution using a retry mechanism, we return a well known error. We can use that error later to decide if we want to requeue the reconciliation or not.